lib/trivial.nix

```diff
@@ -1,18 +1,36 @@
 { lib }:
 
 let
-  inherit (lib.trivial)
```

Comment:
Here's the implicit recursion.

Reply:
I think it qualifies as explicit recursion, because we're pulling from an attrset instead of a `let` binding or `rec` attrset scope.

I'd prefer to inherit from `lib` or `lib.trivial` instead of `builtins`, because inheriting from `lib` instead of `builtins` is a good practice:
 - You won't accidentally depend on a Nix evaluator version that's more recent than the minimum supported by Nixpkgs, `lib/minver.nix`.
   Whoever added the builtin to `lib` (and their reviewer) have done their due diligence.
 - In a few rare cases, the `lib` function has slightly improved behavior. The only instance that comes to mind is `foldl'`'s improved evaluation order.

Reply:
Agreeing with Robert here!